test/20_pa1d_tests.ts

```diff
@@ -536,4 +535,215 @@ describe('PA1D Contract', async function () {
+
+        const sixtyPercentOfRoyalties = contractBalanceBefore.sub(contractBalanceAfter).mul(60).div(100);
+        const fortyPercentOfRoyalties = contractBalanceBefore.sub(contractBalanceAfter).mul(40).div(100);
+
+        expect(contractBalanceAfter).to.be.lt(contractBalanceBefore);
+        expect(Number(accountABalanceAfter)).to.be.approximately(
+          Number(sixtyPercentOfRoyalties.add(accountABalanceBefore)),
+          Number(errorMargin)
+        );
+        expect(Number(accountBBalanceAfter)).to.be.approximately(
+          Number(fortyPercentOfRoyalties.add(accountBBalanceBefore)),
+          Number(errorMargin)
+        );
+      });
+
+      it('should be able to withdraw balance of an ERC20 token', async () => {
+        const ERC20 = await l1.holographErc20.attach(l1.sampleErc20Holographer.address);
+        const SAMPLEERC20 = await l1.sampleErc20.attach(l1.sampleErc20Holographer.address);
+
+        await SAMPLEERC20.mint(owner.address, totalRoyalties);
+        await ERC20.transfer(pad1d.address, totalRoyalties);
+
+        const contractBalanceBefore = await ERC20.balanceOf(pad1d.address);
+        const accountABalanceBefore = await ERC20.balanceOf(anyAddress);
+        const accountBBalanceBefore = await ERC20.balanceOf(notOwner.address);
+
+        const data = (await pad1d.populateTransaction.getTokenPayout(ERC20.address)).data || '';
+        const tx = await l1.factory.connect(owner).adminCall(pad1d.address, data);
+        await tx.wait();
+
+        const accountABalanceAfter = await ERC20.balanceOf(anyAddress);
+        const accountBBalanceAfter = await ERC20.balanceOf(notOwner.address);
+        const contractBalanceAfter = await ERC20.balanceOf(pad1d.address);
+
+        const sixtyPercentOfRoyalties = contractBalanceBefore.sub(contractBalanceAfter).mul(60).div(100);
+        const fortyPercentOfRoyalties = contractBalanceBefore.sub(contractBalanceAfter).mul(40).div(100);
+
+        expect(contractBalanceAfter).to.be.lt(contractBalanceBefore);
+        expect(accountABalanceAfter).to.be.equal(accountABalanceBefore.add(sixtyPercentOfRoyalties));
+        expect(accountBBalanceAfter).to.be.equal(accountBBalanceBefore.add(fortyPercentOfRoyalties));
+      });
+    });
+
+    describe('A collection has 3 recipients with a 20 % / 50% / 30% split', () => {
+      const totalRoyalties = ethers.utils.parseEther('10');
+      const mockAddress = createRandomAddress();
+
+      before(async () => {
+        const data =
+          (
+            await pad1d.populateTransaction.configurePayouts(
+              [anyAddress, notOwner.address, mockAddress],
+              [2000, 5000, 3000]
+            )
+          ).data || '';
+        const tx = await l1.factory.connect(owner).adminCall(pad1d.address, data);
+        await tx.wait();
+      });
+
+      it('should be able to withdraw all native token balance', async () => {
+        await owner.sendTransaction({
+          to: pad1d.address,
```

Comment:
this pa1d thing always leads to typos. we're going to rename this contract from PA1D to Royalties so then we can update these references too